app/src/layouts/kanban/index.ts

```diff
@@ -0,0 +1,596 @@
+			});
+
+			const groupFieldsToLoad = computed(() => {
+				if (primaryKeyField.value === null || groupTitle.value === null) return [];
+				return [primaryKeyField.value?.field, groupTitle.value];
+			});
+
+			const groupTitleFields = computed(() => {
+				if (isRelational.value) {
+					return groupsCollectionFields.value.filter((field) => field.type === 'string' || field.type === 'text');
+				}
+
+				return null;
+			});
+
+			const {
+				items: relationalGroupsItems,
+				loading: groupsLoading,
+				error: groupsError,
+				changeManualSort: groupsChangeManualSort,
+				getItems: getGroups,
+			} = useItems(groupsCollection, {
+				sort,
+				limit: ref(100),
+				page: ref(1),
+				fields: groupFieldsToLoad,
+				filter: ref({}),
+				search: ref(null),
+			});
+
+			const groups = computed(() => {
+				if (isRelational.value) return relationalGroupsItems.value;
+				return selectedGroup.value?.meta?.options?.choices as Record<string, any>[];
+			});
+
+			return {
+				groups,
+				groupsLoading,
+				groupsError,
+				groupsChangeManualSort,
+				info,
+				fields,
+				groupTitleFields,
+				groupsPrimaryKeyField,
+				groupsSortField,
+				groupsCollection,
+				addGroup,
+				editGroup,
+				deleteGroup,
+				changeGroupTitle,
+				changeGroupSort,
+				isRelational,
+			};
+
+			async function deleteGroup(id: string | number) {
+				const pkField = primaryKeyField.value?.field;
+
+				if (pkField === undefined) return;
+
+				items.value = items.value.filter((item) => item[pkField] !== id);
+
+				await api.delete(`/items/${groupsCollection.value}/${id}`);
+
+				await getGroups();
+			}
+
+			async function addGroup(title: string) {
+				if (groupTitle.value === null) return;
+
+				await api.post(`/items/${groupsCollection.value}`, {
+					[groupTitle.value]: title,
+				});
+
+				await getGroups();
+			}
+
+			async function editGroup(id: string | number, title: string) {
+				if (groupTitle.value === null) return;
+
+				await api.patch(`/items/${groupsCollection.value}/${id}`, {
+					[groupTitle.value]: title,
+				});
+
+				await getGroups();
+			}
+
+			async function changeGroupTitle(id: string | number, newTitle: string) {
+				const pkField = primaryKeyField.value?.field;
+
+				if (groupTitle.value === null || pkField === undefined) return;
+
+				const itemIndex = items.value.findIndex((item) => item[pkField] === id);
+
+				const item = items.value[itemIndex];
+
+				if (item) {
+					item[groupTitle.value] = newTitle;
+				}
+
+				await api.patch(`/items/${groupsCollection.value}/${id}`, {
+					[groupTitle.value]: newTitle,
+				});
+
+				await getGroups();
+			}
+
+			async function changeGroupSort(event: ChangeEvent<Group>) {
+				if (event.moved && sortField.value !== null) {
+					await groupsChangeManualSort({
+						item: items.value[event.moved.oldIndex - (showUngrouped.value ? 1 : 0)]?.id,
+						to: items.value[event.moved.newIndex - (showUngrouped.value ? 1 : 0)]?.id,
+					});
+				}
```

Comment:
```suggestion
				if (!event.moved || sortField.value === null) return;
				const offset = showUngrouped.value ? 1 : 0;
				await groupsChangeManualSort({
					item: items.value[event.moved.oldIndex - offset]?.id,
					to: items.value[event.moved.newIndex - offset]?.id,
				});
```

Reply:
Seems like the logic itself had some issues (eg. it should use `groupedItems` instead of `items`, and check `groupsSortField` instead of `sortField`), so I've updated it in https://github.com/directus/directus/pull/18516/commits/94d8e5091532099cca36cdb4a7d517d56e1560c9.

I've applied the suggested code style change in https://github.com/directus/directus/pull/18516/commits/9987cd46125a4711cd8010e2a2b73420bccfcf24 👍